### PR TITLE
[WIP] Integration tests

### PR DESCRIPTION
**Pull request by patriksvensson**
This is currently a work in progress where I try to add the initial code for our integration tests. Expect things to change until this is ready.

#### To skip building of Cake before running tests

````
./tests/integration/build.ps1 -SkipBuildingCake
```

This require Cake to be present somewhere under `./build/`.

#### To run all tests

````
./tests/integration/build.ps1
```

#### To run a single test

````
./tests/integration/build.ps1 -Target "Cake.Common.IO.DirectoryAliases.CreateDirectory"
```

### CODE REVIEW

File: tests/integration/Cake.Common/ArgumentAliases.cake [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+//////////////////////////////////////////////////////////////////////////////
+
+Task("Cake.Common.ArgumentAliases.HasArgument")
+    .Does(() =>
+{
+    // Given, When
+    var arg = Argument<string>("customarg");
+
+    // Then
+    Assert.Equal("hello", arg);
```
Comment (gep13) [inline tests/integration/Cake.Common/ArgumentAliases.cake]: @patriksvensson this is going to fail, right?

**Comment by gep13** [comment]
@patriksvensson this looks really nice! Very clean! I like it! :smile:

File: tests/integration/Cake.Common/ArgumentAliases.cake (hunk 1 above)
Comment (patriksvensson) [inline tests/integration/Cake.Common/ArgumentAliases.cake]: No, this is passing. I pass the `--customarg=hello` to the Cake.exe call in the bootstrapper.

Comment (patriksvensson) [inline tests/integration/Cake.Common/ArgumentAliases.cake]: But as I write this, I realize the Test should be called `Cake.Common.ArgumentAliases.Argument` :smile:

**Pull request closed without merging**